GDJS/Runtime/variable.js

```diff
@@ -4,21 +4,36 @@
  * This project is released under the MIT License.
  */
 
+/**
+ * @typedef variableData Data representation of a GDevelop variable
+ * @property {string} [name] The name of the variable. Used if a child variable.
+ * @property {string | number} [value] The value of the variable, either string or number. Leave blank for structures.
+ * @property {variableData} [children] The children of the structure. Leave blank if value is defined.
+ */
+
 /**
  * A Variable is an object storing a value (number or a string) or children variables.
  *
  * @memberof gdjs
  * @class Variable
- * @param {Object} varData optional object used to initialize the variable.
+ * @param {Object} varData Optional object used to initialize the variable.
+ * @param {Array<variableData>} varData.value A representation of a variable.
  */
 gdjs.Variable = function(varData)
 {
-    this._value = 0;
-    this._str = "";
-    this._numberDirty = false;
-    this._stringDirty = true;
+	/** @type {number} */
+	this._value = 0;
+	/** @type {string} */
+	this._str = "";
+	/** @type {boolean} */
+	this._numberDirty = false;
+	/** @type {boolean} */
+	this._stringDirty = true;
+	/** @type {boolean} */
 	this._isStructure = false;
-    this._children = {};
+	/** @type {Object} */
```

Comment:
Let's be a bit more precise: `@type Object.<string, gdjs.Variable>` - will help for type checking :)

Reply:
fixed